### PR TITLE
[MAINTENANCE] Ruff `0.5.3`

### PR DESCRIPTION
**Pull request by Kilo59**
Update to `ruff` `0.5.3` and enable Github Actions annotations on linter errors.

https://docs.astral.sh/ruff/integrations/#github-actions

image

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -153,6 +153,8 @@ def lint(
         cmds.append("--unsafe-fixes")
     if watch:
         cmds.append("--watch")
+    if os.getenv("GITHUB_ACTIONS"):
```
Comment (billdirks) [inline tasks.py]: Is this always set in github? It allows integration with their UI?

File: tests/datasource/fluent/test_invalid_datasource.py [hunk 2]
```diff
@@ -50,29 +50,28 @@
...
 
     @pytest.mark.parametrize("base_ds_method_name", DATASOURCE_PUBLIC_METHODS)
     def test_datasource_methods(self, base_ds_method_name: str):
         """Ensure that InvalidDatasource overrides the applicable Datasource methods."""
-        for base_ds_method_name in DATASOURCE_PUBLIC_METHODS:
-            method = getattr(InvalidDatasource, base_ds_method_name, None)
-            assert method, f"Expected {base_ds_method_name} to be defined on InvalidDatasource"
-            with pytest.raises(TypeError):
-                method()
+        method = getattr(InvalidDatasource, base_ds_method_name, None)
```
Comment (billdirks) [inline tests/datasource/fluent/test_invalid_datasource.py:63]: Why do we remove the loop over `DATASOURCE_PUBLIC_METHODS` here? Oh just saw the paramterization above and I'm guessing that's the reason? Nice find!

**Review by billdirks — APPROVED** [review]

File: tests/datasource/fluent/test_invalid_datasource.py (hunk 2 above)
Comment (Kilo59) [reply tests/datasource/fluent/test_invalid_datasource.py:63]: Yeah, one of the new ruff rules that caught it and it still took me a bit to see it.

File: tasks.py (hunk 1 above)
Comment (Kilo59) [reply tasks.py]: Yes exactly.
https://docs.astral.sh/ruff/integrations/#github-actions

**Pull request merged**